src/lib/common/sol-missing.h

```diff
@@ -70,6 +70,50 @@ memmem(const void *haystack, size_t haystacklen, const void *needle, size_t need
 }
 #endif
 
+#if !defined(HAVE_PIPE2) && defined(O_CLOEXEC)
+#include <fcntl.h>
+#include <unistd.h>
+#include <sol-util-file.h>
+
+static inline int
+pipe2(int pipefd[2], int flags)
+{
+    int ret = pipe(pipefd);
+    if (ret < 0)
+        return ret;
+
+    if (flags & O_NONBLOCK) {
+        ret = sol_util_fd_set_flag(pipefd[0], O_NONBLOCK);
+        if (ret >= 0)
+            ret = sol_util_fd_set_flag(pipefd[1], O_NONBLOCK);
+        if (ret < 0)
+            goto err;
+    }
+
+    if (flags & O_CLOEXEC) {
+        ret = fcntl(pipefd[0], F_GETFL);
```

Comment:
why `fcntl()` here and `sol_util_fd_set_flag()` above? also, can't we just set them both at once?